### DISCUSSION TITLE
[android] Hermes compatibility

### DISCUSSION THREAD

**Discussion by sabin10** [invertase/react-native-firebase#5424] [General]
Hello! I have some questions about Hermes & react-native-firebase that leads me to confusion. We've recently updated to the 0.64 version of react-native and I rechecked rnfirebase docs about [Hermes support ](https://rnfirebase.io/#hermes-support) so I was very glad to see there is a walkaround (until 0.64 docs were clear that Hermes is not supported).
Then we start testing with Hermes enabled, the app was clearly faster on most of the devices (Android only, didn't try iOS). 

But then we found a small number of devices where the app was slower(TTI was 3-5x slower, delays, lag), we had **Hermes enabled + rnfirebase** on this configuration. After I tried a version where I **removed all rnfirebase configs but keep Hermes enabled**, it was a lot better, leading me to the conclusion that I found some devices where Hermes isn't supported by rnfirebase.

PS: We tried on the latest react-native-firebase versions (12.0.0 and 12.1.0)

We can replicate it on 2 different android models. What can we do now? Would help if we give more info about the buggy devices?

**Comment by mikehardy** [comment]
slower != not supported
crashes == not supported

as far as I know there are no crashes with Hermes enabled on both android and iOS now, so we may say it is supported

Hermes is and always will be a performance optimization though, and performance optimizations always dependent on all inputs plus execution environment

What you found is not that hermes is not supported, but that on certain devices when you include react-native-firebase there is something about those devices (perhaps limited RAM in combo with lots more javascript?) that causes poor performance

What you can do if you like is figure out exactly why Hermes is slower on those devices, and fix it either in Hermes (working with them) or disable hermes

react-native-firebase does nothing special with hermes though, and performance optimization in hermes contexts is something we'll need interested community members to work through with all the involved repositories

**Reply by sabin10** [reply]
Thank you for the explanations. I will take it to Hermes, maybe related https://github.com/facebook/hermes/issues/471#issuecomment-820123463

also seems like RAM was limited on the poor-performance device as the performance continues to decrease when using more background tasks(e.g. having more apps in bg, listening to music)

**Reply by Nantris** [reply]
@sabin10 do you know if performance was improved as hoped for by upgrading to >=0.65.0?
